programs/demo_common.sh

```diff
@@ -0,0 +1,89 @@
+## Common shell functions used by demo scripts programs/*/*.sh.
+
+## How to write a demo script
+## ==========================
+##
+## Include this file near the top of each demo script:
+##   . "${0%/*}/../demo_common.sh"
+##
+## As the last thing in the script, call the cleanup function.
+##
+## You can use the functions and variables described below.
+
+set -e -u
+
+## $root_dir is the root directory of the Mbed TLS source tree.
```

Comment:
Suggestion for improvement: explain why we need to guess at it. (I think it's because this will be included from scripts that might be at various depths from the root, but it took me a few minutes to think of that, and I'm not even sure.)

Reply:
Note: I wonder it this is really worth it. Since each demo script already needs to know the relative path to this, it also knows its distance to the root. We could get rit of this guessing if we changed the usage instructions to:
```
programs_dir="${0%/*}"; programs_dir="${programs_dir%/*}" # repeated depending on depth
source "$programs_dir/demo_common.sh"
```
Wdyt?

Reply:
Doesn't work if the script is invoked from its own directory. I wanted demo scripts to be somewhat robust, and in particular to run from any directory.

The scripts assume an in-tree build. It would be nice to distinguish the build directory from the source directory. Maybe in a follow-up.